Hw19/ManchkinGame/Fight.cpp

```diff
@@ -0,0 +1,88 @@
+
+#include "Fight.h"
+
+#include <algorithm>
+
+#include "Item.h"
+#include "ItemDeck.h"
+#include "Modifier.h"
+#include "Munchkin.h"
+#include "Monster.h"
+#include "Runaway.h"
+#include "Victory.h"
+#include "ModifierDeck.h"
+
+void Fight::start()
+{
+	m_result = FightResult::InProgress;
+
+	calculateMunchkinPower();
+	calculateMonsterPower();
+}
+
+void Fight::applyModifier(int choice)
+{
+	Modifier* modifier = m_munchkin->popModifier(choice);
+	if (modifier == nullptr)
+	{
+		throw std::exception();
+		return;
+	}
+
+	modifier->apply(m_munchkin, m_monster);
+
+	m_MunchkinPower += modifier->getMunchkinModifier();
+	m_monsterPower += modifier->getMonsterModifier();
+}
+
+void Fight::runawayFlow()
+{
+	Runaway* policy = m_monster->getRunawayPolicy();
+	if(policy)
+	    policy->apply(m_munchkin);
+
+	m_result = FightResult::MonsterWon;
+}
+
+void Fight::victoryFlow(ModifierDeck *modifierDeck, ItemDeck *itemsDeck)
+{
+	
+	m_munchkin->updateLevelBy(1);
+
+	m_munchkin->addItem(itemsDeck->generateItem());
+
+	Victory *policy = m_monster->getVictoryPolicy();
+	if(policy)
+	{
+		
+		const size_t addCards = policy->apply(m_monster);
```

Comment:
you assume here that Victory policies can ONLY addCards. But the main purpose of the base class is to support any policies implemented in children objects, not only additional cards